store/types/validity.go

```diff
@@ -1,15 +1,30 @@
 package types
 
-// AssertValidKey checks if the key is valid(key is not nil)
+import math "math"
+
+const (
+	// 32K
+	MaxKeyLength = math.MaxUint16
+	// 4G
+	MaxValueLength = math.MaxUint32
+)
+
+// AssertValidKey checks if the key is valid(key is not nil, not empty and within length limit)
 func AssertValidKey(key []byte) {
 	if len(key) == 0 {
-		panic("key is nil")
+		panic("key is nil or empty")
+	}
+	if len(key) > MaxKeyLength {
+		panic("key is too large")
 	}
 }
 
-// AssertValidValue checks if the value is valid(value is not nil)
+// AssertValidValue checks if the value is valid(value is not nil and within length limit)
 func AssertValidValue(value []byte) {
 	if value == nil {
 		panic("value is nil")
 	}
+	if len(value) > MaxValueLength {
+		panic("value is too large")
+	}
 }
```

Comment:
Valid functions like these need to return errors rather than panicking, so that callers can detect when the validation checks have failed. Otherwise, there's no reason to have these functions at all, because any access of invalid values would trigger panics, anyway.